Ticket6465 error on crisp and surf

LowriJenkins opened the pull request:
https://github.com/ISISComputingGroup/IBEX/issues/6465
CRISP, SURF, and INES were erroring due to 2 different problems.
SURF's issue has been resolved by adding cryptography to the requirments of the database populator.
CRISP and INES were erroring due to having no data. It has now been made more clear when lack of data is the reason that the database populator fails.

to test:
run the databse populator in debug mode, on CRISP, SURF, and INES. Enusring that SURF runs properly, and CRISP and INES give the following error `[ERROR] Unable to update INSTRUMENT_NAME, no data found.`

Commit:
updated gatherer to fail more clearly when there is no data

Commit:
updated requirements to contain cryptography to prevent error

DominicOram reviewed (changes requested):
Works well but a few of things:
* It breaks one of the unit tests
* Can you add a new unit test to confirm that this does happens
* I think in this case we want to update the database with no data, otherwise there could be left over old data

Commit:
Fixed failing test, and added test for empty skipping update call

LowriJenkins commented:
> 
> 
> Works well but a few of things:
> 
>     * It breaks one of the unit tests
> 
>     * Can you add a new unit test to confirm that this does happens
> 
>     * I think in this case we want to update the database with no data, otherwise there could be left over old data

I've fixed the failing test and added a new one, on the update with no data, would it be fine to call the function that removes expired data (i.e. things past 100 days) directly, since updateing directly will attempt to unpack the data that doesn't exist?

Commit:
updated so that expired data will still be removed even if no new data

Commit:
fixed accidental duplicate test

DominicOram commented:
Sure, that will work

DominicOram committed:
Minor tidying up

DominicOram reviewed (approved):
Thanks!
